h2o-algos/src/main/java/hex/ensemble/StackedEnsembleModel.java

```diff
@@ -529,11 +519,11 @@ private Frame getLevelOnePredictFrame(Frame fr, Frame adaptFrm, Job j) {
       transform = null;
     }
     final String seKey = this._key.toString();
-    final Key<Frame> levelOneFrameKey = Key.make("preds_levelone_" + seKey + fr._key);
-    Frame levelOneFrame = transform == null ?
-            new Frame(levelOneFrameKey)  // no transform -> this will be the final frame
-            :
-            new Frame();        // transform -> this is only an intermediate result
+    final String frId = ""+(fr._key == null ? fr.checksum() : fr._key);
```

Comment:
@tomasfryda I had true difficulties figuring out why test `pyunit_plot_functions__add_saving_parameter_and_decorate_plot_result.py` was consistently failing with NPE (see http://mr-0xg1:8080/job/h2o-3-pipeline/job/PR-6335/32/testReport/junit/(root)/r_suite/Py3_6_Small___pyunit_plot_functions__add_saving_parameter_and_decorate_plot_result_py/)

I figured out that there must have been some race condition somewhere when `fr` as a `null` key (which is the case when `score` is called from `PartialDependence.CalculatePdpPerBin`): apparently we're building several levelOnePredict frames in parallel in that case (maybe you can shed me some light on this, as I don't see where the parallelism is done) so if the `levelOneFrame` key is not unique, then nasty things may happen (in the test, I guess one frame was emptied when getting out of scope at the same time that another one was getting transformed…).

Please consider this change and if you see why this race condition was not previously occuring (or maybe it was but with different consequence when using a wrong frame in some circumstances)

Reply:
regarding the parallelism, I'm stupid, `PartialDependence` calls
```java
CalculatePdpPerBin pdp = ...
fs.add(H2O.submitTask(pdp)); 
```
in a loop multiple times, so that's where the parallelism occurs and therefore the risks of race condition.

Reply:
^^ fixed in last commit